subprojects/maven/src/main/java/org/gradle/api/publish/maven/tasks/PublishToMavenRepository.java

```diff
@@ -91,11 +104,20 @@ private PublishSpec computeSpec() {
             throw new InvalidUserDataException("The 'publication' property is required");
         }
 
-        MavenArtifactRepository repository = getRepository();
+        DefaultMavenArtifactRepository repository = this.repository.get();
         if (repository == null) {
             throw new InvalidUserDataException("The 'repository' property is required");
         }
 
+        Property<Credentials> credentialsProperty = repository.getConfiguredCredentials();
```

Comment:
💄 This whole credentials safety check could be its own method.

Reply:
```
    private void checkCredentialSafety(DefaultMavenArtifactRepository repository) {
        Property<Credentials> credentialsProperty = repository.getConfiguredCredentials();
        Credentials value = credentialsProperty.getOrNull();

        boolean safeCredentials = value == null || areCredentialsSafe(repository.getName(), value);

        if (!safeCredentials) {
            getListenerManager().getBroadcaster(CredentialListener.class).onUnsafeCredentials("repository " + repository.getName(), this);
        }
    }
    ```